### ISSUE TITLE
Make block snapping and selection non recursive

### ISSUE THREAD

**Issue by robinp7720** [robinp7720/WebBloks#5]
Browsers limit function recursion and just makes the code harder to understand.

**Comment by an-OK-squirrel** [comment]
I'm having the same problem over at elem :P

**Comment by robinp7720** [comment]
Ok, I've figured how I'm going to do it :p
But I'm gonna have to rewrite most of the block rendering.

I made one using html elements without recursion in about 10 mins?

**Comment by an-OK-squirrel** [comment]
Oh wow. I haven't really tried tbh ;p

**Comment by robinp7720** [comment]
Its much easier with html elements because can easily target an element by looping through each element with a .block tag and check if the cursor is anywhere on that block.

Thats what I'm trying to do with this now. I'm trying to store blocks inside of a blocks placed object so I can loop through each block with a simple for loop. each block object inside of the blocks placed then has a parent id and a script id. I might remove the script id though. I would still need to find the position of each block, which is alot easier if your using html elements, by tracing back the block chain to the script object.

Using a canvas can be much harder for some things :p

**Comment by an-OK-squirrel** [comment]
:p

**Comment by matthewr6** [comment]
Is this still being worked on?

**Comment by robinp7720** [comment]
@Firedrake969 Yes, I'm also on holidays/moving to another country..

**Comment by an-OK-squirrel** [comment]
dang

**Comment by an-OK-squirrel** [comment]
Dang to moving to another country, not that you're still working on it :P

**Comment by matthewr6** [comment]
@an-OK-squirrel haha
also this is why we use HTML and have a team :P

**Comment by robinp7720** [comment]
@Firedrake969 This is really a project where I'm trying to learn javascript and see if I can actually do this with a canvas". I am using html, This is a web app :p
Html elements cant do as much as you can do with the canvas :p

**Comment by matthewr6** [comment]
Heh.  We've had this discussion before many times.  We'll see ;)

**Comment by robinp7720** [comment]
@Firedrake969 Elemental looks fine as it is. You basicly have the editor done as I see it. Just the blocks are missing :p 

If you want to use a canvas, you'll have to completely redo elemental :p

**Comment by matthewr6** [comment]
Exactly.  GL, HF, and don't try to be competitive.  It's kinda pointless under these circumstances, IMO.

**Comment by robinp7720** [comment]
@Firedrake969 I'm not trying to be.
I'm trying to make WebBlox a web editor. Not a learning tool.

**Comment by matthewr6** [comment]
So good luck with that - since it'll have to start with the basics anyways :)

**Comment by an-OK-squirrel** [comment]
yeah xD

TBH, I'd like to see some form of elemental extension system, so you can do something like this.

Anyways, good luck and stuff!

**Comment by matthewr6** [comment]
@an-OK-squirrel still have to have the basics - don't even worry about new features until the absolute core is done

**Comment by matthewr6** [comment]
@robinp7720 what do you mean by "resurrective"?

**Comment by robinp7720** [comment]
@Firedrake969 A function calling itself. \* Recursive, not resurrective :p
It can lead to performance issues and most browsers limit the amount a function can call itself.

**Comment by matthewr6** [comment]
Oh, better choice of words (though you kind of have to have recursion in any powerful code xD)  It's not really the issue of functions calling themselves but rather how long the JS takes to run (a function calling itself 1000 times will perform about as well as a loop that loops 1000 times - if you make sure the function calls itself only on the first iteration, of course).  Infinite loops are never good, and sloppy recursiveness in any style (loops, functions, etc) is just a bad idea.

**Comment by matthewr6** [comment]
@robinp7720 still active?

**Comment by robinp7720** [comment]
@Firedrake969 I though your making elemental? :p And most browsers limit the amount of recursion that can happen. Now I'm adding everything to a json object and going through and rendering each item linearly.

**Comment by matthewr6** [comment]
35k isn't that bad, and you can't really go through trees without recursion xD  Just curious... your sense of being "competitive" helps give me ideas :package:

**Comment by robinp7720** [comment]
@Firedrake969 This is only a side project..

**Comment by matthewr6** [comment]
same with mine xD

**Comment by robinp7720** [comment]
@Firedrake969 So, whats your "main" project then?

**Comment by matthewr6** [comment]
Proprietary stuff.

**Comment by robinp7720** [comment]
@Firedrake969 For whom...?

**Comment by matthewr6** [comment]
Can't say or it'd give my location away.  Feel free to not believe me though :P
Anyways, that's irrelevant.

**Comment by an-OK-squirrel** [comment]
You saying you alre

**Comment by an-OK-squirrel** [comment]
Dy have a job?

**Comment by matthewr6** [comment]
maybe

**Comment by an-OK-squirrel** [comment]
wat 

darnit

**Comment by matthewr6** [comment]
huh

you have things to in the issues :P  does anybody besides me use waffle?  :P

**Comment by an-OK-squirrel** [comment]
huh?

**Comment by matthewr6** [comment]
check.
gitter.

**Comment by an-OK-squirrel** [comment]
k
if you want to know why I'm not working on elemental...

**Comment by matthewr6** [comment]
?

**Comment by matthewr6** [comment]
we should stop chatting on ~~an abandoned~~ a project that's not ours xD

**Comment by robinp7720** [comment]
@Firedrake969 I hate :heart: you too

**Comment by matthewr6** [comment]
wat
changed wording

**Comment by robinp7720** [comment]
@Firedrake969, your manners are among the worst I have seen. Your attitude is also insanely disrespectful.

**Comment by an-OK-squirrel** [comment]
...

**Comment by matthewr6** [comment]
says the guy who said "I hate :heart: you" or something of the sort
I'm sorry if you misconstrue your actions and my reactions for my behavior.

so that said
not going to bug the owner of an old project any more :P

**Comment by an-OK-squirrel** [comment]
You both are fine, unless you're around eachother. Are you brothers?

**Comment by matthewr6** [comment]
> not going to bug the owner of an old project any more :P

**Comment by robinp7720** [comment]
@an-OK-squirrel This would look alot different if we were..

**Comment by an-OK-squirrel** [comment]
k

**Comment by an-OK-squirrel** [comment]
you were what?

**Comment by robinp7720** [comment]
... If we were brothers ...

**Comment by matthewr6** [comment]
and this is why we never got anything done for a while...

seriously guys just drop it

**Comment by an-OK-squirrel** [comment]
yeah

**Comment by an-OK-squirrel** [comment]
I'm fine with some things, but "I hate you" is just too far ...

**Comment by robinp7720** [comment]
The disrespect from @Firedrake969 is also too far

**Comment by an-OK-squirrel** [comment]
every action has an equal and opposite reaction

'cuz that makes sense

**Comment by robinp7720** [comment]
@an-OK-squirrel 
@Firedrake969 
There is no reason for you to be disrespectful here.
If you two want to stay here, then be constructive.
If you are incapable of being constructive, then I'd recommend you to leave

**Comment by an-OK-squirrel** [comment]
have you solved the problem yet?

**Comment by matthewr6** [comment]
I'm not disrespectful :)  I'm just responding to your behavior accordingly.
I'll leave only on my own terms or if it is forced by locking this issue.  That is all.
